subprojects/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/GradleEnterprisePluginCheckInIntegrationTest.groovy

```diff
@@ -121,8 +124,36 @@ class GradleEnterprisePluginCheckInIntegrationTest extends AbstractIntegrationSp
         getMinimumPluginVersionForConfigurationCaching() | true
     }
 
+    @Requires(IntegTestPreconditions.NotConfigCached)
+    def "shows warning message when Gradle Enterprise plugin version is used with isolated projects enabled"() {
+        given:
+        plugin.runtimeVersion = plugin.artifactVersion = pluginVersion
+        applyPlugin()
+        settingsFile << """
+            println "present: " + services.get($GradleEnterprisePluginManager.name).present
+        """
+
+        when:
+        succeeds("t", "-D${StartParameterBuildOptions.IsolatedProjectsOption.PROPERTY_NAME}=true")
+
+        then:
+        output.contains("present: ${applied}")
+
+        and:
+        output.contains("gradleEnterprisePlugin.checkIn.unsupported.reasonMessage = $UNSUPPORTED_PLUGIN_DUE_TO_ISOLATED_PROJECTS_MESSAGE") != applied
```

Comment:
💅 Do you think changing the assertion to something like this would be more readable?

```
!applied || output.contains("...")
```

Reply:
I think this is not quite similar with the current version. Because if `applied == false` then we don't checking second part of condition.

Current version is verifying also `if plugin was applied then output doesn't contains unsupported message`

Reply:
You are right, nevermind 👍